src/test/java/com/example/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,208 @@
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+    }
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) throws SQLException {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/books/insert-test-books.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/users/insert-test-users-with-shopping-carts.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/cartitems/insert-test-cartitem.sql"));
+        }
+    }
+
+    @AfterEach
+    void tearDown(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/cartitems/delete-all-cartitems.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/users/delete-all-users.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/delete-all-test-books.sql"));
+        }
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("""
+            Verify getById method works
+            """)
+    void getById_ValidId_ReturnsShoppingCartDto() throws Exception {
+        //given
+        Long id = 1L;
+        ShoppingCartDto expected = getTestShoppingCartDto();
+        User user = getTestUser();
+
+        //when
+        MvcResult result = mockMvc.perform(get("/cart")
+                        .with(user(user))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //then
+        ShoppingCartDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                ShoppingCartDto.class);
```

Comment:
dont use object mapper, spring boot test has `jsonpath` to simplify\optimize your tests
```suggestion;
```